core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -902,8 +902,13 @@ private[spark] object MapOutputTracker extends Logging {
     // the contents don't have to be copied to the new buffer.
     val out = new ApacheByteArrayOutputStream()
     out.write(DIRECT)
-    val codec = CompressionCodec.createCodec(conf, "zstd")
-    val objOut = new ObjectOutputStream(codec.compressedOutputStream(out))
+    val shouldCompress = conf.get(MAP_STATUS_COMPRESS)
+    val codec = if (shouldCompress) {
+      Some(CompressionCodec.createCodec(conf, conf.get(IO_COMPRESSION_CODEC.key, "zstd")))
```

Comment:
This is a regression, @Ngone51 . IO_COMPRESSION_CODEC is `lz4` by default.

Reply:
hmmm... actually, I'm indicated to use `zstd` by default in this way.  There's difference between `conf.get(IO_COMPRESSION_CODEC.key, "zstd")` and `conf.get(IO_COMPRESSION_CODEC)`, right ? Or something wrong with the former one ?